packages/params/src/forkName.ts

```diff
@@ -20,28 +20,61 @@ export enum ForkSeq {
   deneb = 4,
 }
 
+function exclude<T>(coll: T[], val: T[]): T[] {
+  return coll.filter((f) => !val.includes(f));
+}
+
+export function highestFork(forkNames: ForkName[]): ForkName {
+  let highest = forkNames[0];
+
+  for (const forkName of forkNames) {
+    if (ForkSeq[forkName] > ForkSeq[highest]) {
+      highest = forkName;
+    }
+  }
+
+  return highest;
+}
+
+export function lowestFork(forkNames: ForkName[]): ForkName {
+  let lowest = forkNames[0];
+
+  for (const forkName of forkNames) {
+    if (ForkSeq[forkName] < ForkSeq[lowest]) {
+      lowest = forkName;
+    }
+  }
+
+  return lowest;
+}
+
 export type ForkAll = ForkName;
+export const forkAll = Object.keys(ForkName) as ForkName[];
 
 export type ForkPreLightClient = ForkName.phase0;
 export type ForkLightClient = Exclude<ForkName, ForkPreLightClient>;
+export const forkLightClient = exclude(forkAll, [ForkName.phase0]);
 export function isForkLightClient(fork: ForkName): fork is ForkLightClient {
   return fork !== ForkName.phase0;
 }
 
 export type ForkPreExecution = ForkPreLightClient | ForkName.altair;
 export type ForkExecution = Exclude<ForkName, ForkPreExecution>;
+export const forkExecution = exclude(forkAll, [ForkName.phase0, ForkName.altair]);
```

Comment:
can these have their types narrowed? and then highestFork and lowestFork can be generic?

eg:
```ts
export const forkExecution = ... as ForkExecution[];

export function highestFork<F>(forkNames: F[]): F;

const b: BeaconBlock<ForkExecution> = sszTypesFor(highestFork(forkExecution), "BeaconBlock");
```